ambry-store/src/main/java/com.github.ambry.store/PersistentIndex.java

```diff
@@ -625,16 +694,18 @@ FindInfo findEntriesSince(FindToken token, long maxTotalSizeOfEntries) throws St
         // check journal
         List<JournalEntry> entries = journal.getEntriesSince(offsetToStart, storeToken.getInclusive());
         logger.trace("Journal based token, Time used to get entries: {}", (time.milliseconds() - startTimeInMs));
+        ConcurrentSkipListMap<Offset, IndexSegment> indexes = indexesRef.get();
```

Comment:
can you confirm something. We take the snapshot of the index right here. There are chances that indexes change after this and the following lines are going to consider the snapshot only. We are still sure we don't miss out on any entries, bcoz new entries will definitely be part of journal and doesn't affect the correctness. Hope you also have thought about it. Can you confirm my understanding and there are no issues